lib/galaxy/model/migrate/versions/0136_record_workflow_outputs.py

```diff
@@ -0,0 +1,180 @@
+    )
+
+    workflow_invocation_step_table = Table(
+        "workflow_invocation_step", metadata,
+        Column("id", Integer, primary_key=True),
+        Column("create_time", DateTime, default=now),
+        Column("update_time", DateTime, default=now, onupdate=now),
+        Column("workflow_invocation_id", Integer, ForeignKey("workflow_invocation.id"), index=True, nullable=False),
+        Column("workflow_step_id", Integer, ForeignKey("workflow_step.id"), index=True, nullable=False),
+        Column("action", JSONType, nullable=True),
+        Column("state", TrimmedString(64), default="new"),
+    )
+
+    workflow_invocation_step_job_association_table = Table(
+        "workflow_invocation_step_job_association", metadata,
+        Column("id", Integer, primary_key=True),
+        Column("workflow_invocation_step_id", Integer, ForeignKey("workflow_invocation_step.id"), index=True, nullable=False),
+        Column("index", Integer, nullable=True),
+        Column("job_id", Integer, ForeignKey("job.id"), index=True, nullable=False),
+    )
+
+    tables = {
+        "workflow_invocation_output_dataset_association": workflow_invocation_output_dataset_association_table,
+        "workflow_invocation_output_dataset_collection_association": workflow_invocation_output_dataset_collection_association_table,
+        "workflow_invocation_step_output_dataset_association": workflow_invocation_step_output_dataset_association_table,
+        "workflow_invocation_step_output_dataset_collection_association": workflow_invocation_step_output_dataset_collection_association_table,
+        "workflow_invocation_step": workflow_invocation_step_table,
+        "workflow_invocation_step_job_association": workflow_invocation_step_job_association_table,
+    }
+
+    return tables
+
+
+def upgrade(migrate_engine):
+    metadata.bind = migrate_engine
+    print(__doc__)
+
+    LegacyWorkflowInvocationStep_table = Table("workflow_invocation_step", metadata, autoload=True)
+    ExistingWorkflowInvocation_table = Table("workflow_invocation", metadata, autoload=True)
+
+    cons = ForeignKeyConstraint([LegacyWorkflowInvocationStep_table.c.workflow_invocation_id], [ExistingWorkflowInvocation_table.c.id])
+    cons.drop()
+
+    for index in LegacyWorkflowInvocationStep_table.indexes:
+        index.drop()
+
+    LegacyWorkflowInvocationStep_table.rename("worklfow_invocation_step_premigrate135")
+    # Try to deregister that workflow_invocation_step to work around some caching problems
+    # it seems.
+    LegacyWorkflowInvocationStep_table.deregister()
+    metadata._remove_table("workflow_invocation_step", metadata.schema)
+
+    metadata.reflect()
+    tables = get_new_tables()
+    for table in tables.values():
+        __create(table)
+
+    def nextval(table, col='id'):
+        if migrate_engine.name in ['postgres', 'postgresql']:
+            return "nextval('%s_%s_seq')" % (table, col)
+        elif migrate_engine.name in ['mysql', 'sqlite']:
```

Comment:
Remove `'sqlite'` from here, I suppose.

Reply:
I removed it from the other case - I think null was the right thing to do all along - I was tracking down the wrong NULL constraint.